phoenix-core/src/it/java/org/apache/phoenix/end2end/ExplainPlanWithStatsEnabledIT.java

```diff
@@ -47,46 +53,116 @@
 import org.apache.phoenix.util.EnvironmentEdge;
 import org.apache.phoenix.util.EnvironmentEdgeManager;
 import org.apache.phoenix.util.PhoenixRuntime;
-import org.junit.BeforeClass;
+import org.apache.phoenix.util.PropertiesUtil;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
 
+import com.google.common.base.Optional;
 import com.google.common.collect.Lists;
 
 /**
  * This class has tests for asserting the bytes and rows information exposed in the explain plan
  * when statistics are enabled.
  */
+@RunWith(Parameterized.class)
 public class ExplainPlanWithStatsEnabledIT extends ParallelStatsEnabledIT {
 
+    private static final long STATS_UPDATE_FREQUENCY = 5l;
     private static String tableA;
     private static String tableB;
     private static String tableWithLargeGPWidth;
     private static String indexOnA;
     private static final long largeGpWidth = 2 * 1000 * 1000L;
 
-    @BeforeClass
-    public static void createTables() throws Exception {
-        tableA = generateUniqueName();
-        initDataAndStats(tableA, 20L);
-        tableB = generateUniqueName();
-        initDataAndStats(tableB, 20L);
-        tableWithLargeGPWidth = generateUniqueName();
-        initDataAndStats(tableWithLargeGPWidth, largeGpWidth);
-        indexOnA = generateUniqueName();
-        createIndex(indexOnA, tableA, 20);
+    private static final List<String> GUIDE_POSTS_CACHE_FACTORY_LIST =
+            Lists.newArrayList(
+                    "org.apache.phoenix.query.SingletonProviderGuidePostsCacheFactory",
+                    "org.apache.phoenix.query.DefaultGuidePostsCacheFactory");
+
+    @Parameters
+    public static Iterable<? extends String> guidePostsCacheFactoryStrings(){
+        return GUIDE_POSTS_CACHE_FACTORY_LIST;
+    }
+
+    private String cacheFactoryString;
+
+    public ExplainPlanWithStatsEnabledIT(String cacheFactoryString){
+        this.cacheFactoryString = cacheFactoryString;
+    }
+
+    private boolean init = false;
```

Comment:
Do you need this variable? init() is called only once, isn't it?

Reply:
Forget about this comment. I saw you need this variable to make sure it is initialized only once when we use Parameterized.class.